server/mvc/controllers/ProfileController.js

```diff
@@ -1,16 +1,27 @@
-var session = require('express-session');
 var bnet = require('battlenet-api')(process.env.BNET_ID);
+var Models = require('../models.js');
 
 exports.info = function(req, res){
   if(!req.isAuthenticated()) {
     res.redirect('/');
     return;
   }
-  /*
-  TODO: Implement using "Models.User"
-  bnet.account.sc2({origin: 'eu', access_token: session.OAuthAccessToken}, function(bnetErr, bnetResp){
-    res.send(JSON.stringify(bnetResp));
-  });*/
+  var user = new Models.User();
+  user
+    .findByUserId(req.user.id)
+    .then(function(){
+      if (!user.exists()) {
+        throw new Error('User does not exit');
+      }
+      var oauthTokenEncryptionSalt = process.env.BNET_OAUTH_TOKEN_ENCRYPTION_SALT;
+      var oauthToken = user.decryptOauthToken(user.getValue('oauthTokenEncrypted'), oauthTokenEncryptionSalt);
+      bnet.account.sc2({origin: 'eu', access_token: oauthToken}, function(bnetErr, bnetResp){
```

Comment:
Just to make sure we're on the same page: Maybe here we should update the Profile model with what we get from Battle.net (or have an update button and present what's in the db instead of fetching each time).